SOURCES/etc/xapi.d/plugins/linstor

```diff
@@ -0,0 +1,188 @@
+
+            host_devices = ','.join(disks[hostname])
+            session.xenapi.host.call_plugin(
+                host_ref, LVM_PLUGIN, 'create_physical_volume', {
+                    'devices': host_devices,
+                    'force': str(force)
+                }
+            )
+
+            session.xenapi.host.call_plugin(
+                host_ref, LVM_PLUGIN, 'create_volume_group', {
+                    'vg_name': volume_group,
+                    'devices': host_devices
+                }
+            )
+
+            if thin:
+                session.xenapi.host.call_plugin(
+                    host_ref, LVM_PLUGIN, 'create_thin_pool', {
+                        'vg_name': volume_group,
+                        'lv_name': THIN_POOL
+                    }
+                )
+
+            # Create SR.
+            master_ref = session.xenapi.pool.get_all_records().values()[0]["master"]
+
+            device_config = {
+                'redundancy': redundancy,
+                'provisioning': 'thin' if thin else 'thick',
+                'group-name': '{}/{}'.format(volume_group, THIN_POOL) if thin else volume_group,
+                'hosts': ','.join(hostnames),
+                'monitor-db-quorum': str(len(hostnames) > 2)
+            }
+            sr_ref = session.xenapi.SR.create(
+                master_ref, device_config, '0', name, description, 'linstor', '', True, {}
+            )
+            return session.xenapi.SR.get_uuid(sr_ref)
+
+    finally:
+        session.xenapi.session.logout()
+
+def _list_drbd_volumes(volume_group=None):
+    drbd_volumes = {}
+    result = run_command(['drbdsetup', 'show', '--json'])
+    if result['exit']:
+        raise Exception(result)
+
+    config = json.loads(result['stdout'])
+    for resource in config:
+        for volume in resource['_this_host']['volumes']:
+            backing_disk = volume['backing-disk']
+            match = BACKING_DISK_RE.match(backing_disk)
+            if not match:
+                continue
+
+            cur_volume_group = match.groups()[0]
+            if volume_group and cur_volume_group != volume_group:
+                continue
+
+            minor = int(volume['device_minor'])
+            if cur_volume_group in drbd_volumes:
+                drbd_volumes[cur_volume_group].append(minor)
+            else:
+                drbd_volumes[cur_volume_group] = [minor]
+    return drbd_volumes
+
+def _destroy_drbd_volume(minor):
+    result = run_command(['drbdsetup', 'detach', minor, '--force'])
+    if result['exit']:
+        raise Exception(result)
+    result = run_command(['drbdsetup', 'del-minor', minor])
+    if result['exit']:
+        raise Exception(result)
+
+# ------------------------------------------------------------------------------
+
+@error_wrapped
+def is_installed(session, args):
+    # TODO: Check packages.
+    pass
+
+@error_wrapped
+def install(session, args):
+    # TODO: Install packages from repo.
+    pass
```

Comment:
you can see how `netdata` plugin does it.

Reply:
Unrelated. The packages are not in our repo, :wink: